File: only-buns-be/src/main/java/com/example/onlybunsbe/DTO/UserDTO.java
package com.example.onlybunsbe.DTO;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class UserDTO {
    private Long id;
    private String email;
    private String username;
    private String firstName;
    private String lastName;
    private String address;
    private boolean isActive;
    private LocalDateTime registeredAt;
    private String role;
    private int postCount;         // Broj objava
    private int followingCount;    // Broj korisnika koje prati
    private int followerCount;     // Broj pratilaca

    private List<Long> sentMessageIds;    // ID-evi poslatih poruka
    private List<Long> groupIds;          // ID-evi grupa u kojima je korisnik član
}

File: only-buns-fe/src/app/dto/user.dto.ts
export interface UserDTO {
  id: number;
  email: string;
  username: string;
  firstName: string;
  lastName: string;
  address?: string;
  isActive: boolean;
  registeredAt: Date;
  role: string;
  postCount: number;        // Broj objava
  followingCount: number;    // Broj korisnika koje prati
  followerCount: number;     // Broj pratilaca
}
